official/resnet/resnet_run_loop.py

```diff
@@ -72,6 +75,16 @@ def process_record_dataset(dataset, is_training, batch_size, shuffle_buffer,
   # dataset for the appropriate number of epochs.
   dataset = dataset.repeat(num_epochs)
 
+  if is_training and num_gpus is not None and num_gpus > 0:
+    total_examples = num_epochs * examples_per_epoch
+    # Force the number of batches to be divisible by the number of devices.
+    # This prevents some devices from receiving batches while others do not,
+    # which can lead to a lockup. This case will soon be handled directly by
+    # distribution strategies, at which point this .take() operation will no
+    # longer be needed.
+    total_batches = total_examples // batch_size // num_gpus * num_gpus
+    dataset.take(total_batches * batch_size)
```

Comment:
To consider: if we calculate this number in the fn that calls this one, we don't have to pass num_gpus and total_examples so far in, and can just pass an ... examples_to_use or better-named equivalent through to here.

Reply:
I considered this, but I like keeping the pipeline logic together and passing run info one layer deeper was an acceptable tradeoff.